### PR TITLE
Fixes some pickup behaviours

### PR DESCRIPTION
**Pull request by PowerfulBacon**
## About The Pull Request

Adds in a new flag indicating when objects have been picked up and calls pickup() on item equip to hand if this flag is not set. The flag is unset when dropped or thrown.

## Why It's Good For The Game

Previously pickup() wasn't called properly on all thinsg, and certain things that put an item directly into your hand did not call pickup() which could bypass or break the behaviours of certain objects. Stuff like package wrapping was affected by this which could allow you to do things like pickup supermatter silvers without tongs, pickup items such as clockwork hammers etc.

## Changelog
:cl:
fix: Fixes pickup() not being called when items were directly forced into your hands.
/:cl:

### CODE REVIEW

File: code/__DEFINES/obj_flags.dm [hunk 1]
```diff
@@ -31,6 +31,7 @@
 #define IMMUTABLE_SLOW			(1<<10) //! When players should not be able to change the slowdown of the item (Speed potions, etc)
 #define IN_STORAGE				(1<<11) //! is this item in the storage item, such as backpack? used for tooltips
 #define ILLEGAL					(1<<12)	//! this item unlocks illegal tech
+#define PICKED_UP				(1<<13)	//! This item has been picked up by someone
```
Comment (Archanial) [inline code/__DEFINES/obj_flags.dm]:
```suggestion
#define PICKED_UP				(1<<13)	//! This item has been picked up by someone, prevents calling pickup proc on item until it is dropped
```

Comment (PowerfulBacon) [inline code/__DEFINES/obj_flags.dm]:
```suggestion
#define PICKED_UP				(1<<13)	//! This item has been picked up by someone and is on their person. Prevents pickup() from being called when an item is moved to a player's hand from the backpack, and allows pickup() to be called when the item is picked up from a non-player location (IE when something is moved into the players hand by force, or by them clicking on something)
```

Comment (PowerfulBacon) [reply code/__DEFINES/obj_flags.dm]: oh didn't see that

File: code/game/objects/items.dm [hunk 2]
```diff
@@ -631,6 +630,7 @@ GLOBAL_VAR_INIT(rpg_loot_items, FALSE)
 /obj/item/proc/pickup(mob/user)
```
Comment (PowerfulBacon) [inline code/game/objects/items.dm:672]:
```suggestion
/obj/item/proc/pickup(mob/user)
	SHOULD_CALL_PARENT(TRUE)
```

**Comment by PowerfulBacon** [comment]
done

**Comment by Archanial** [comment]
> files changed 8
done
> files changed 70

🦖

File: code/game/objects/items.dm [hunk 3]
```diff
@@ -217,7 +217,7 @@ GLOBAL_VAR_INIT(rpg_loot_items, FALSE)
 		item_flags |= IN_STORAGE
 
 	if(istype(loc, /obj/item/robot_module))
-		item_flags |= IN_INVENTORY
+		pickup()
```
Comment (Archanial) [inline code/game/objects/items.dm]: shouldn't you be passing user here?

Comment (PowerfulBacon) [reply code/game/objects/items.dm]: yea

Comment (Archanial) [reply code/game/objects/items.dm]: ok

**Comment by ike709** [comment]
@PowerfulBacon

File: code/game/objects/items.dm (hunk 3 above)
Comment (Archanial) [reply code/game/objects/items.dm]:
```suggestion
		pickup(user)
```

**Review by Archanial — DISMISSED** [review]

**Comment by Archanial** [comment]
This is a proper maintainer moment

**Review by Archanial — DISMISSED** [review]

File: code/modules/clothing/head/soft_caps.dm [hunk 4]
```diff
@@ -10,9 +10,9 @@
 	var/flipped = 0
 
 /obj/item/clothing/head/soft/dropped()
-	icon_state = "[soft_type]soft"
-	flipped=0
 	..()
+	src.icon_state = "[item_color]soft"
```
Comment (PowerfulBacon) [inline code/modules/clothing/head/soft_caps.dm]:
```suggestion
	src.icon_state = "[soft_type]soft"
```

File: code/modules/clothing/head/soft_caps.dm [hunk 5]
```diff
@@ -10,9 +10,9 @@
...
+	src.flipped=0
```
Comment (Archanial) [inline code/modules/clothing/head/soft_caps.dm]:
```suggestion
	icon_state = "[soft_type]soft"
	flipped=0
```

**Review by ike709 — APPROVED** [review]

**Pull request merged**